torchaudio/functional.py

```diff
@@ -424,7 +424,8 @@ def create_fb_matrix(
         f_min: float,
         f_max: float,
         n_mels: int,
-        sample_rate: int
+        sample_rate: int,
+        norm: str = "",
```

Comment:
As a public API signature, I think `Optional[str]` looks cleaner.

Reply:
This was made due to [comment](https://github.com/pytorch/audio/pull/589#discussion_r420324932). I'll leave it as it is for now. We can always extend the `str` to `Optional[str]` without BC breaking later :)

Reply:
It looks to me that, that comment was meant for the type of variable to pass when running Torchscript test, not about the function signature.

Reply:
If we allow `None` in the signature, then the code should work with/without jit when passing `None`. It wasn't though. Is that what you meant?

Reply:
> If we allow None in the signature, then the code should work with/without jit when passing None.
Yes, it works.

```
from typing import Optional

import torch
from torch import Tensor


def bar(foo: Optional[str]=None) -> Tensor:
    if foo is None:
        return torch.zeros(1, 2)
    if foo == "a":
        return torch.ones(1, 1)

    return torch.empty(1, 1)


ts_bar = torch.jit.script(bar)

for v in [None, "a", "b"]:
    print(bar(v))
    print(ts_bar(v))
```

produces

```
None
tensor([[0., 0.]])
tensor([[0., 0.]])
a
tensor([[1.]])
tensor([[1.]])
b
tensor([[-2.8910e+12]])
tensor([[0.]])
```

also `dcshift` uses `Optional[float]` and it works fine. for both None and float input for torchscript. 
https://github.com/pytorch/audio/pull/558

Reply:
It's just when type is optional, it firstly needs to compare against `None` using `if var is None` or `if var is not None`.

Reply:
https://pytorch.org/docs/master/jit_language_reference.html#optional-type-refinement

Reply:
Alrighty, #641 :)